src/main/java/org/opensearch/ad/model/ADTask.java

```diff
@@ -613,185 +368,40 @@ public static ADTask parse(XContentParser parser, String taskId) throws IOExcept
-                detector,
-                detectionDateRange,
-                entity,
-                parentTaskId,
-                estimatedMinutesLeft,
-                user
-            );
-    }
-
-    public String getTaskId() {
-        return taskId;
-    }
-
-    public void setTaskId(String taskId) {
-        this.taskId = taskId;
-    }
-
-    public Instant getLastUpdateTime() {
-        return lastUpdateTime;
-    }
-
-    public String getStartedBy() {
-        return startedBy;
-    }
-
-    public String getStoppedBy() {
-        return stoppedBy;
-    }
-
-    public String getError() {
-        return error;
-    }
-
-    public void setError(String error) {
-        this.error = error;
-    }
-
-    public String getState() {
-        return state;
-    }
-
-    public void setState(String state) {
-        this.state = state;
-    }
-
-    public String getId() {
-        return detectorId;
-    }
-
-    public Float getTaskProgress() {
-        return taskProgress;
-    }
-
-    public Float getInitProgress() {
-        return initProgress;
-    }
-
-    public Instant getCurrentPiece() {
-        return currentPiece;
-    }
-
-    public Instant getExecutionStartTime() {
-        return executionStartTime;
-    }
-
-    public Instant getExecutionEndTime() {
-        return executionEndTime;
-    }
-
-    public Boolean getLatest() {
-        return isLatest;
-    }
-
-    public String getTaskType() {
-        return taskType;
-    }
-
-    public String getCheckpointId() {
-        return checkpointId;
+        int superHashCode = super.hashCode();
+        int hash = Objects.hashCode(configId, detector, detectionDateRange);
+        hash += 89 * superHashCode;
+        return hash;
     }
 
     public AnomalyDetector getDetector() {
         return detector;
     }
 
-    public String getCoordinatingNode() {
-        return coordinatingNode;
-    }
-
-    public String getWorkerNode() {
-        return workerNode;
-    }
-
     public DateRange getDetectionDateRange() {
         return detectionDateRange;
     }
 
-    public Entity getEntity() {
-        return entity;
-    }
-
-    public String getEntityModelId() {
-        return entity == null ? null : entity.getModelId(getId()).orElse(null);
-    }
-
-    public String getParentTaskId() {
-        return parentTaskId;
-    }
-
-    public Integer getEstimatedMinutesLeft() {
-        return estimatedMinutesLeft;
-    }
-
-    public User getUser() {
-        return user;
-    }
-
     public void setDetectionDateRange(DateRange detectionDateRange) {
         this.detectionDateRange = detectionDateRange;
     }
 
-    public void setLatest(Boolean latest) {
-        isLatest = latest;
-    }
-
-    public void setLastUpdateTime(Instant lastUpdateTime) {
-        this.lastUpdateTime = lastUpdateTime;
+    @Override
+    public String getEntityModelId() {
+        return entity == null ? null : entity.getModelId(getConfigId()).orElse(null);
```

Comment:
Why is there a `getConfigID` here but in forecasting its just `.getModelId(configID)`

Reply:
I am gonna replace getConfigId() with configID to make it consistent.

Reply:
sweet, and then it can be refactored to TimeSeriesTask I assume

Reply:
yes